pact/marmalade.repl

```diff
@@ -1041,6 +1041,197 @@
+
+(env-sigs
+  [{'key:'mint
+   ,'caps:
+    [(marmalade.ledger.MINT "project-3" "bob" 1.0)
+     ]}])
+
+(expect "bob mints project-3"
+  "Write succeeded"
+  (mint 'project-3 "bob" (read-keyset 'bob-guard) 1.0))
+
+(expect "Bob is credited"
+  1.0
+  (get-balance 'project-3 'bob))
+
+(expect "Create and mint fqp token EVENTS"
+  [ {"name": "marmalade.ledger.TOKEN","params": ["project-3"]}
+    {"name": "marmalade.ledger.MINT","params": ["project-3" "bob" 1.0]}
+    {"name": "marmalade.ledger.SUPPLY","params": ["project-3" 1.0]}]
+    (map (remove 'module-hash) (env-events true)))
+
+(commit-tx)
+
+(begin-tx "sale fixed-quote-royalty-policy success")
+(env-hash (hash "fixed-quote-royalty-policy-sale-tx2"))
+(use marmalade.ledger)
+
+(env-data {"alice": {"keys": ["alice"], "pred":"="}})
+(expect
+  "Create recipient account alice"
+  "Write succeeded"
+  (coin.create-account 'alice (read-keyset 'alice)))
+
+(env-sigs
+  [{'key:'bob
+   ,'caps:
+    [(marmalade.ledger.OFFER "project-3" "bob" 0.2 120)]}
+ ])
+
+(env-data
+  {
+   "quote": {
+       "price": 0.00000000001
+      ,"recipient": "alice"
+      ,"recipient-guard": {"keys": ["alice"], "pred":"="}
+    }
+  ,"buyer": "coin-buyer"
+  ,"buyer-guard": {"keys": ["buyer"], "pred": "="}
+  })
+(env-chain-data {"block-height": 100})
+
+(test-capability (coin.COINBASE))
+
+(expect "coinbase buyer" "Write succeeded"
+  (coin.coinbase "coin-buyer" (read-msg 'buyer-guard) 2.0))
+
+(expect "coin-buyer is credited coins"
+  2.0
+  (coin.get-balance 'coin-buyer))
+
+(expect-failure
+  "0.00000000000006"
+  (sale 'project-3 'bob 0.2 120))
+
+(env-data
+  {
+   "quote": {
+       "price": 1.0
+      ,"recipient": "alice"
+      ,"recipient-guard": {"keys": ["alice"], "pred":"="}
+    }
+  ,"buyer": "coin-buyer"
+  ,"buyer-guard": {"keys": ["buyer"], "pred": "="}
+  })
+
+(expect "Sale succeeds"
+  true
+  (sale 'project-3 'bob 0.2 120))
+
+(expect "coin balance of coin-buyer is unchanged"
+  2.0
+  (coin.get-balance 'coin-buyer))
+
+(expect "Seller is credited, sale account is debited on project-3 balance"
+  [0.8 0.2]
+  (map (get-balance 'project-3) ['bob (sale-account)]))
+
+(expect "OFFER events"
+   [{"name": "marmalade.ledger.OFFER","params": ["project-3" "bob" 0.2 120]}
+    {"name": "marmalade.ledger.SALE","params": ["project-3" "bob" 0.2 120 (pact-id)]}
+    {"name": "marmalade.ledger.TRANSFER","params": ["project-3" "bob" (sale-account) 0.2]}]
+  (map (remove 'module-hash) (env-events true)))
+
+
+(expect-failure "Withdraw fails before timeout"
+  "still active"
+  (continue-pact 0 true))
+
+(env-sigs [
+  {'key: 'bob
+   ,'caps: [(marmalade.ledger.BUY "project-3" "bob" "coin-buyer" 0.2 120 (hash "fixed-quote-royalty-policy-sale-tx2"))]
+  }
+,{'key: 'dummy
+ ,'caps: [
+   (coin.TRANSFER "coin-buyer" "alice" 0.194)
+   (coin.TRANSFER "coin-buyer" "creator" 0.006)
+ ]}])
+
+(expect-failure "buy fails, buyer doesn't sign coin.TRANSFER"
+  "[buyer]"
```

Comment:
hmm that's weird error text, what's going on there?

Reply:
Error text is from guard failure at `coin.TRANSFER`, where buyer keyset is `{"keys": ["buyer"], "pred": "="}`